src/server/scripts/World/boosted_xp.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2008-2017 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "GameTime.h"
+#include "ScriptMgr.h"
+#include "World.h"
+
+#include <boost/date_time/posix_time/posix_time.hpp>
+
+namespace
+{
+    bool IsXPBoostActive()
+    {
+        auto now = boost::posix_time::to_tm(boost::posix_time::from_time_t(GameTime::GetGameTime()));
```

Comment:
I wonder if a return early case like "if (weekdayMaskBoosted == 0) return false;" will avoid checking every time for gametime and weekday.

Reply:
Hmm, not sure both of these changes are needed. Just either one.
During runtime the optimization in IsXPBoostActive would be just an overhead as the script is already disabled in scriptloader if the script is off. Though I guess adding a single IF statement probably isnt a huge bottleneck..

Reply:
yeah, the check in AddWorldScripts() is enough